components/brave_today/browser/brave_news_controller.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_TODAY_BROWSER_BRAVE_NEWS_CONTROLLER_H_
+#define BRAVE_COMPONENTS_BRAVE_TODAY_BROWSER_BRAVE_NEWS_CONTROLLER_H_
+
+#include <memory>
+#include <string>
+
+#include "base/callback_forward.h"
+#include "base/containers/flat_map.h"
+// #include "base/synchronization/lock.h"
+#include "base/one_shot_event.h"
+#include "base/timer/timer.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_today/browser/feed_controller.h"
+#include "brave/components/brave_today/browser/publishers_controller.h"
+#include "brave/components/brave_today/common/brave_news.mojom-forward.h"
+#include "brave/components/brave_today/common/brave_news.mojom.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "mojo/public/cpp/bindings/receiver_set.h"
+#include "mojo/public/cpp/bindings/remote.h"
+
+class PrefService;
+
+namespace brave_ads {
+class AdsService;
+}  // namespace brave_ads
+
+namespace user_prefs {
+class PrefRegistrySyncable;
+}  // namespace user_prefs
+
+namespace history {
+class HistoryService;
+}  // namespace history
+
+namespace brave_news {
+
+// Browser-side handler for Brave News mojom API, 1 per profile
+class BraveNewsController : public KeyedService,
+                            public mojom::BraveNewsController {
+ public:
+  // static void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable*
```

Comment:
Can you remove this too? (if it's not needed; if needed, definitely need to implement 😛)